x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.tsx

```diff
@@ -46,45 +60,62 @@ export const heatmapTitle = i18n.translate('xpack.maps.source.esGridHeatmapTitle
   defaultMessage: 'Heat map',
 });
 
-export class ESGeoGridSource extends AbstractESAggSource {
-  static type = SOURCE_TYPES.ES_GEO_GRID;
-
-  static createDescriptor({ indexPatternId, geoField, metrics, requestType, resolution }) {
+export class ESGeoGridSource extends AbstractESAggSource implements ITiledSingleLayerVectorSource {
+  static createDescriptor(
+    descriptor: Partial<ESGeoGridSourceDescriptor>
+  ): ESGeoGridSourceDescriptor {
+    descriptor = AbstractESAggSource.createDescriptor(descriptor);
+    if (typeof descriptor.indexPatternId !== 'string') {
+      throw new Error('Cannot create ESGeoGridSource descriptor without indexPatternId');
+    }
+    if (!descriptor.requestType) {
```

Comment:
Don't think you need to throw for missing `requestType`, just default to clusters